src/templates/globals/wrapper/index.tsx

```diff
@@ -1,20 +1,50 @@
-import { Banner } from '@/templates/globals/banner'
+import { useState, useEffect } from 'react'
+import { Banner } from '@/templates/globals/banners/banner'
+import { PromoBanner } from '@/templates/globals/banners/promoBanner'
+import { FacilityBanner } from '@/templates/globals/banners/facilityBanner'
 import { isEmpty } from 'lodash'
+import { BannerType, PromoBannerType, FacilityBannerType } from '@/types/index'
+import { BannerDisplayType, BannerTypeMapping } from '@/data/queries/banners'
+
 export interface LayoutProps {
-  children?: React.ReactNode
   props?: any
-  bannerData?: any
+  children?: React.ReactNode
+  bannerData?: BannerType[] | PromoBannerType[] | FacilityBannerType[]
 }
 
-export function Wrapper({ props, bannerData, children }: LayoutProps) {
-  if (isEmpty(bannerData)) bannerData = props?.bannerData
+export const formatBannerType = (bannerData) => {
+  switch (bannerData?.type as string) {
+    case BannerTypeMapping[BannerDisplayType.PROMO_BANNER]:
+      return <PromoBanner key={bannerData.id} {...bannerData} />
+    case BannerTypeMapping[BannerDisplayType.FACILITY_BANNER]:
+      return <FacilityBanner key={bannerData.id} {...bannerData} />
+    case BannerTypeMapping[BannerDisplayType.BANNER]:
+      return <Banner key={bannerData.id} {...bannerData} />
+    default:
+      return null
+  }
+}
+
+export function Wrapper({ bannerData, children }: LayoutProps) {
+  const [showBanners, setShowBanners] = useState(false)
+
+  useEffect(() => {
+    if (isEmpty(bannerData)) {
+      return setShowBanners(false)
+    }
+    if (bannerData) {
+      bannerData?.map(
+        (banner) => formatBannerType(banner) && setShowBanners(true)
+      )
+    }
+  }, [bannerData, showBanners])
```

Comment:
Refactored the banner display to us a useEffect and useState hook to monitor live updates to banners

Reply:
More cleanup added